deauthalyzer.py

```diff
@@ -28,9 +29,15 @@ def check_root_privileges():
 def get_wifi_interfaces():
     interfaces = psutil.net_if_addrs()
     wifi_interfaces = []
-    for interface, addresses in interfaces.items():
-        if interface.startswith('wlan'):
```

Comment:
we can change this line to `if interface.startswith('wlan') or interface.startswith('wlp'):`  as per the new naming convention of the network interfaces in ubuntu, the first two letter describe the type of interface. 'wl' for wlan, 'en' for ethernet. The following code is often a description of the physical placement of the device in your computer - p2 is likely [PCI](https://en.wikipedia.org/wiki/Conventional_PCI) bus 2, and s0 is likely slot 0. So this change should work

Reply:
hey @z0m31en7, you are correct, a simple check would already solve the problem, the network interface may vary according to the nomenclature of network interfaces depending on the system.

- For me, the correct thing would be to check the primary interface, instead of using `or`.
- We would just change `wlan` to `wl`. With that, the script would set the first valid network interface.

```python
def get_wifi_interfaces():
    interfaces = psutil.net_if_addrs()
    wifi_interfaces = []
    for interface, _ in interfaces.items():
        if interface.startswith('wl'):
           wifi_interfaces.append(interface)

    return wifi_interfaces
```
As you yourself mentioned, `wl` for wlan interfaces, in the above part of the code, the `startswith('wl')` function takes the first valid wlan interface, regardless of the network interface naming.

**OBS:** I haven't tested this logic on kali linux for example, but I believe it will solve the problem. inform me :)